test/unit/services/finance/billing_service_test.rb

```diff
@@ -80,6 +80,35 @@ class Finance::BillingServiceTest < ActionDispatch::IntegrationTest
     finished = true
   end
 
+  test 'raises unless a single buyer is billed and charged' do
+    opts = [
+      [5], # account == provider which means to bill a whole provider
+      [6, { provider_account_id: 6 }], # again account == provider
+      [nil], # neither provider nor a buyer specified
+    ]
+
+    opts.each do |args|
+      bs = Finance::BillingService.new(*args)
+      assert_raise(Finance::SpuriousBillingError) do
+        bs.call
+      end
+    end
+  end
```

Comment:
I think it's better to split this one into three, so if one of them fails, we can see it easily in Circle CI
```suggestion
  opts = [
    [5], # account == provider which means to bill a whole provider
    [6, { provider_account_id: 6 }], # again account == provider
    [nil], # neither provider nor a buyer specified
  ]

  opts.each do |args|
    test 'raises unless a single buyer is billed and charged with args: #{args}' do
      bs = Finance::BillingService.new(*args)
      assert_raise(Finance::SpuriousBillingError) do
        bs.call
      end
    end
  end
```

Reply:
I know this. But to me it's more clear to read in this way. Once we run locally, I expect it to be easy to reproduce in case ever this fails.

I'm very disappointed by minitest's lack of `assert_all` and perhaps general development of the project. You can see my discussion here: minitest/minitest#948

I even have the idea to try make rails switch to test-unit or at least optionally when I have time. Although that might also be never.